helper_methods.rb

```diff
@@ -33,71 +32,73 @@ def aspace_staging_login()
   @client = ArchivesSpace::Client.new(@config).login
 end
 
-def get_all_resource_records_for_institution(resolve = [])
-  #run through all repositories (1 and 2 are reserved for admin use)
-  repos_all = (3..12).to_a
-  resources_endpoints = repos_all.map do |repo|
-    'repositories/'+repo.to_s+'/resources'
-    end
-  #debug
-  #puts "endpoints to process are:"
-  #puts resources_endpoints
+def get_all_repo_uris
+  repositories = @client.get('/repositories').parsed
+  repositories.map {|repo| repo['uri']}
+end
 
-  #for each endpoint, get the count of records
-  @results = []
-  resources_endpoints.each do |endpoint|
-    @ids_by_endpoint = []
-    @ids_by_endpoint << @client.get(endpoint, {
+def get_repo_id_from_uri(uri)
+  uri.gsub('/repositories/', '')
+end
+
+def get_resource_ids_for_all_repos
+  repos = get_all_repo_uris
+  @ids = []
+  repos.each do |repo|
+    @ids << @client.get("#{repo}/resources", {
       query: {
-       all_ids: true
+        all_ids: true
       }}).parsed
-    @ids_by_endpoint = @ids_by_endpoint.flatten!
-    count_ids = @ids_by_endpoint.count
-    #debug
-    #puts "number of records to retrieve for #{endpoint}:"
-    #puts count_ids
-
-    #for each endpoint, get the record by id and add to array of records
-    paginate_endpoint(@ids_by_endpoint, count_ids, endpoint, resolve)
-  end #close resources_endpoints.each
-
-  #return array of results
-  @results = @results.flatten!
-end #close method
+  end
+  @ids.flatten!
+end
 
-def get_all_event_records_for_institution(resolve = [])
-  #run through all repositories (1 and 2 are reserved for admin use)
-  repos_all = (3..12).to_a
-  resources_endpoints = repos_all.map do |repo|
-    'repositories/'+repo.to_s+'/events'
+def get_resource_uris_for_all_repos
+  @uris = get_all_repo_uris.map do |repo|
+    ids = @client.get("#{repo}/resources", {
+    query: {
+      all_ids: true
+    }}).parsed
+    ids.map do |id|
+      "#{repo}/resources/#{id}"
     end
-  #debug
-  #puts "endpoints to process are:"
-  #puts resources_endpoints
+  end
+  @uris.flatten!
```

Comment:
Does `uris` need the `@`?

Also, if this exact instance variable `@uris` is not used in another method (that is to say, if other methods only use the return value), you can use `flatten` instead of `flatten!`